metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/ObjectsImporter.java

```diff
@@ -142,9 +162,64 @@ public Analysis importAnalysis(String accession) {
     protected abstract Analysis extractStudyFromAnalysis(AnalysisType analysisType, Analysis analysis);
 
     public ReferenceSequence importReferenceSequence(String accession) {
+        ReferenceSequence referenceSequence = null;
+        try {
+            String assemblyXml = sraXmlRetrieverByAccession.getXml(accession);
+            AssemblyType assembly = sraAssemblyXmlParser.parseXml(assemblyXml, accession);
+            referenceSequence = referenceSequenceConverter.convert(assembly);
+            Taxonomy taxonomy = taxonomyExtractorFromReferenceSequence.getTaxonomy(assembly.getTAXON());
+            referenceSequence.setTaxonomy(taxonomy);
+            referenceSequence = referenceSequenceRepository.findOrSave(referenceSequence);
+        } catch (Exception exception) {
+            IMPORT_LOGGER.log(Level.SEVERE, "Encountered Exception for ReferenceSequence accession " + accession);
+            IMPORT_LOGGER.log(Level.SEVERE, exception.getMessage());
+        }
+        return referenceSequence;
+    }
+
+    private Set<String> getReferenceSequenceAccessions(AnalysisType analysisType) {
+        Set<String> referenceSequenceAccessions = new HashSet<>();
+        AnalysisType.ANALYSISTYPE analysisType1 = analysisType.getANALYSISTYPE();
+        // ENA records can contain a reference sequence in either of the three analysis categories:
+        // REFERENCE_ALIGNMENT, SEQUENCE_VARIATION, and PROCESSED_READS. All accessions are collected and returned.
```

Comment:
The EVA and AMP only deal with SEQUENCE_VARIATION analyses, could @jdylan please confirm if EGA needs others?

Reply:
By the way, should metadata API support ENA as well eventually, or it's not planned, ever?

Reply:
The API doesn't need to support ENA.

Reply:
Will remove `REFERENCE_ALIGNMENT` and `PROCESSED_READS`

Reply:
Has @jdylan confirmed that the EGA is not using them?

Reply:
I interpreted https://github.com/EBIvariation/metadata-ws/pull/50#discussion_r288970412 so. Will wait for formal confirmation from @jdylan

Reply:
I could see from the xmls EGA only uses SEQUENCE_VARIATION. but anyways Dylan can confirm.

Reply:
Need to handle all 3. https://www.ebi.ac.uk/panda/jira/browse/T2D-297?focusedCommentId=281856&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-281856

Reply:
In the latest version of the PR it's implemented slightly different from my original implementation. Originally it combined accessions from all types of analyses if it could find them; now it checks sequentially in a `else-if` chain. If we are guaranteed for sure that two different tags cannot be present in the same analysis (e. g. a `SEQUENCE_VARIATION` and a `REFERENCE_ALIGNMENT`), then it's fine. @cyenyxe @jdylan can you confirm that each analysis contains at most one analysis type?

Reply:
Yes, in today's i confirmed with @jdylan. Analysis can be of only one type SEQUENCE_VARIATION or REFERENCE_ALIGNMENT or PROCESSED_READS

Reply:
In the current implementation all three analysis types are supported in a reasonable manner. Marking as resolved.